<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from 4arms.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="4Armdrone">
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <link name="drone">
    <visual name="drone_visual">
      <origin rpy="1.57079 0 0" xyz="-0.09837 0.046645 0"/>
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/Gazebo_Base_Body.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <origin rpy="1.57079 0 0" xyz="-0.09837 0.046645 0"/>
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/Gazebo_Base_Body.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="arm_1">
    <visual name="arm_1_link_visual">
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/Gazebo_Base_arm.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/Gazebo_Base_arm.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin xyz="0 0 0"/>
    </collision>
  </link>
  <link name="arm_2">
    <visual name="arm_2_link_visual">
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/Gazebo_Base_arm.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin xyz="0 -0.035 0"/>
    </visual>
    <collision>
      <origin xyz="0 -0.035 0"/>
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/Gazebo_Base_arm.STL" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="arm_3">
    <visual name="arm_3_link_visual">
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/Gazebo_Base_arm.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/Gazebo_Base_arm.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin xyz="0 0 0"/>
    </collision>
  </link>
  <link name="arm_4">
    <visual name="arm_4_link_visual">
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/Gazebo_Base_arm.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin xyz="0 -0.035 0"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/Gazebo_Base_arm.STL" scale="0.001 0.001 0.001"/>
      </geometry>
      <origin xyz="0 -0.035 0"/>
    </collision>
  </link>
  <joint name="base_to_arm_1" type="revolute">
    <parent link="drone"/>
    <child link="arm_1"/>
    <origin rpy="0 0 2.1495963267948968" xyz="-0.06906999999999999 0.046645 0"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="1000.0" lower="0.0" upper="1.74" velocity="1000.0"/>
  </joint>
  <joint name="base_to_arm_2" type="revolute">
    <parent link="drone"/>
    <child link="arm_2"/>
    <origin rpy="0 0 -2.1495963267948968" xyz="-0.06906999999999999 -0.046645 0"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="1000.0" lower="0.0" upper="1.74" velocity="1000.0"/>
  </joint>
  <joint name="base_to_arm_3" type="revolute">
    <parent link="drone"/>
    <child link="arm_3"/>
    <origin rpy="0 0 -0.9919963267948966" xyz="0.06906999999999999 -0.046645 0"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="1000.0" lower="0.0" upper="1.74" velocity="1000.0"/>
  </joint>
  <joint name="base_to_arm_4" type="revolute">
    <parent link="drone"/>
    <child link="arm_4"/>
    <origin rpy="0 0 0.9919963267948966" xyz="0.06906999999999999 0.046645 0"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="1000.0" lower="0.0" upper="1.74" velocity="1000.0"/>
  </joint>
  <link name="prop_1">
    <visual name="prop_1_link_visual">
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/1345_prop_ccw.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.03 -0.17 0"/>
    </visual>
  </link>
  <joint name="prop_to_arm_1" type="continuous">
    <parent link="arm_1"/>
    <child link="prop_1"/>
    <origin rpy="0 0 0" xyz="0.155 0.0175 0"/>
    <axis xyz="0.0 0.0 1"/>
  </joint>
  <link name="prop_2">
    <visual name="prop_2_link_visual">
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/1345_prop_cw.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.03 -0.17 0"/>
    </visual>
  </link>
  <joint name="prop_to_arm_2" type="continuous">
    <parent link="arm_2"/>
    <child link="prop_2"/>
    <origin rpy="0 0 0" xyz="0.155 -0.0175 0"/>
    <axis xyz="0.0 0.0 1"/>
  </joint>
  <link name="prop_3">
    <visual name="prop_3_link_visual">
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/1345_prop_ccw.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.03 -0.17 0"/>
    </visual>
  </link>
  <joint name="prop_to_arm_3" type="continuous">
    <parent link="arm_3"/>
    <child link="prop_3"/>
    <origin rpy="0 0 0" xyz="0.155 0.0175 0"/>
    <axis xyz="0.0 0.0 1"/>
  </joint>
  <link name="prop_4">
    <visual name="prop_4_link_visual">
      <geometry>
        <mesh filename="package://urdf_tutorial_r2d2/meshes/1345_prop_cw.STL"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.03 -0.17 0"/>
    </visual>
  </link>
  <joint name="prop_to_arm_4" type="continuous">
    <parent link="arm_4"/>
    <child link="prop_4"/>
    <origin rpy="0 0 0" xyz="0.155 -0.0175 0"/>
    <axis xyz="0.0 0.0 1"/>
  </joint>
</robot>
